lib/streamlit/elements/image.py

```diff
@@ -305,28 +350,28 @@ def image_to_url(
 
 
 def marshall_images(
-    coordinates,
-    image,
-    caption,
-    width,
-    proto_imgs,
-    clamp,
-    channels="RGB",
-    output_format="auto",
-):
-    channels = channels.upper()
+    coordinates: str,
+    image: ImageOrImageList,
+    caption: Optional[Union[str, "npt.NDArray[Any]", List[str]]],
+    width: int,
+    proto_imgs: ImageListProto,
+    clamp: bool,
+    channels: Channels = "RGB",
+    output_format: OutputFormat = "auto",
+) -> None:
+    channels = cast(Channels, channels.upper())
 
     # Turn single image and caption into one element list.
-    if type(image) is list:
+    images: Sequence[AtomicImage]
+    if isinstance(image, list):
         images = image
+    elif type(image) == np.ndarray and len(image.shape) == 4:
```

Comment:
nit: could this also be changed to an `isinstance` check?

Reply:
Done!